integration-tests/paho-mqtt5/src/main/java/org/apache/camel/quarkus/component/paho/mqtt5/it/PahoMqtt5Resource.java

```diff
@@ -60,12 +86,94 @@ public Response producePahoMessage(
+            @PathParam("queueName") String queueName,
+            String message) throws Exception {
+        producerTemplate.sendBodyAndHeader("paho-mqtt5:test?retained=true&brokerUrl=" + brokerUrl("tcp"), message,
+                PahoMqtt5Constants.CAMEL_PAHO_OVERRIDE_TOPIC, queueName);
+        return Response.created(new URI("https://camel.apache.org/")).build();
+    }
+
+    @Path("/readThenWriteWithFilePersistenceShouldSucceed")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String readThenWriteWithFilePersistenceShouldSucceed(@QueryParam("message") String message) throws Exception {
+        producerTemplate.sendBody(
+                "paho-mqtt5:withFilePersistence?retained=true&persistence=FILE&brokerUrl="
+                        + brokerUrl("tcp"),
+                message);
+        return consumerTemplate.receiveBody(
+                "paho-mqtt5:withFilePersistence?persistence=FILE&brokerUrl=" + brokerUrl("tcp"),
+                5000,
+                String.class);
+    }
+
+    @Path("/routeStatus/{id}")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String routeStatus(@PathParam("id") String routeId,
+            @QueryParam("waitForContainerStarted") @DefaultValue("false") boolean wait) throws Exception {
+        RouteController routeController = context.getRouteController();
+        if (wait) {
+            counter.await(30, TimeUnit.SECONDS);
+        }
+        return routeController.getRouteStatus(routeId).name();
+    }
+
+    @Path("/mock")
+    @POST
+    @Produces(MediaType.TEXT_PLAIN)
+    public String mock(String message) throws Exception {
+        MockEndpoint endpoint = context.getEndpoint("mock:test", MockEndpoint.class);
+        endpoint.expectedBodiesReceived(message);
+
+        endpoint.assertIsSatisfied();
+        return "OK";
+    }
+
+    @Path("/send")
+    @POST
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response send(String message) throws Exception {
+        producerTemplate.sendBody("direct:test", message);
         return Response.created(new URI("https://camel.apache.org/")).build();
     }
 
     private String brokerUrl(String protocol) {
         return ConfigProvider.getConfig().getValue("paho5.broker." + protocol + ".url", String.class);
     }
 
+    private void setKeyStore(String keystore, String password) {
+        InputStream in = Thread.currentThread().getContextClassLoader().getResourceAsStream(keystore);
+
+        try {
+            Files.copy(in, Paths.get(keystore));
+        } catch (Exception e) {
+        }
+
+        System.setProperty("javax.net.ssl.keyStore", keystore);
+        System.setProperty("javax.net.ssl.keyStorePassword", password);
+        System.setProperty("javax.net.ssl.trustStore", keystore);
+        System.setProperty("javax.net.ssl.trustStorePassword", password);
+    }
+
+    private void removeKeyStore(String keystore) {
+        try {
+            Files.delete(Paths.get(keystore));
```

Comment:
Should we unset the SSL system properties here?

Reply:
good catch! I will unset these properties.

Reply:
Also, I guess we want this to always happen. E.g even if the call to the `ProducerTemplate` throws an exception.